### DISCUSSION TITLE
alvr AUR packages fail to build

### DISCUSSION THREAD

**Discussion by emilyyyylime** [alvr-org/ALVR#1861] [Q&A]
The build script fails at the following step:
```
$ bash -c "./configure --prefix=/home/emilylime/.cache/paru/clone/alvr/src/alvr/deps/linux/ffmpeg/alvr_build --enable-gpl --enable-version3 --enable-static --disable-programs --disable-doc --disable-avdevice --disable-avformat --disable-swresample --disable-swscale --disable-postproc --disable-network --disable-everything --enable-encoder=h264_vaapi --enable-encoder=hevc_vaapi --enable-hwaccel=h264_vaapi --enable-hwaccel=hevc_vaapi --enable-filter=scale_vaapi --enable-vulkan --enable-libdrm --enable-pic --enable-rpath --enable-encoder=h264_nvenc --enable-encoder=hevc_nvenc --enable-nonfree --enable-cuda-nvcc --enable-libnpp --nvccflags=\"-gencode arch=compute_52,code=sm_52 -O2\" --extra-cflags=\"-I/opt/cuda/targets/x86_64-linux/include\" --extra-ldflags=\"-L/opt/cuda/targets/x86_64-linux/lib\""
ERROR: failed checking for nvcc.

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: command exited with non-zero code `bash -c "./configure --prefix=/home/emilylime/.cache/paru/clone/alvr/src/alvr/deps/linux/ffmpeg/alvr_build --enable-gpl --enable-version3 --enable-static --disable-programs --disable-doc --disable-avdevice --disable-avformat --disable-swresample --disable-swscale --disable-postproc --disable-network --disable-everything --enable-encoder=h264_vaapi --enable-encoder=hevc_vaapi --enable-hwaccel=h264_vaapi --enable-hwaccel=hevc_vaapi --enable-filter=scale_vaapi --enable-vulkan --enable-libdrm --enable-pic --enable-rpath --enable-encoder=h264_nvenc --enable-encoder=hevc_nvenc --enable-nonfree --enable-cuda-nvcc --enable-libnpp --nvccflags=\"-gencode arch=compute_52,code=sm_52 -O2\" --extra-cflags=\"-I/opt/cuda/targets/x86_64-linux/include\" --extra-ldflags=\"-L/opt/cuda/targets/x86_64-linux/lib\""`: 1', alvr/xtask/src/dependencies.rs:190:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'alvr-20.4.3-1': 
error: packages failed to build: alvr-20.4.3-1
```
I have installed `cuda-tools` suspecting that's the issue, but seemingly not. Am I missing anything?

**Comment by emilyyyylime** [comment]
I couldn't even get it to compile with the distrobox setup; this is completely blocking me from being able to use VR on my PC :(

**Comment by FlyvendeMus** [comment]
I get the exact same error. I tried asking Gemini for help and it made me Export the batch of nvcc
`NVCC_PATH=$(dirname $(which nvcc))
export PATH=$PATH:$NVCC_PATH:/opt/cuda/bin`
Did not work

Then made me create a syslink to the correct folder
`sudo ln -s /opt/cuda/bin/nvcc /usr/bin/nvcc`
Did not work

Then it made me edit the PKGBUILD file to add the export path command
`build() {
    cd "$srcdir/alvr"

    # Set ALVR environment variables
    export ALVR_ROOT_DIR=/usr
    export ALVR_LIBRARIES_DIR="$ALVR_ROOT_DIR/lib"
    export ALVR_OPENVR_DRIVER_ROOT_DIR="$ALVR_LIBRARIES_DIR/steamvr/alvr/"
    export ALVR_VRCOMPOSITOR_WRAPPER_DIR="$ALVR_LIBRARIES_DIR/alvr/"
    export FIREWALL_SCRIPT_DIR="$ALVR_ROOT_DIR/share/alvr/"

    # Source profile then FORCE the CUDA paths
    source /etc/profile
    export PATH=$PATH:/opt/cuda/bin
    export CUDA_PATH=/opt/cuda
    export NVCC_APPEND_FLAGS+='-std=c++14'

    # Clean and minimal prepare-deps command
    # Removed --nvidia and --platform to prevent argument errors
    cargo run --release --frozen -p alvr_xtask -- prepare-deps

    # Build the ALVR components
    cargo build \
        --frozen \
        --release \
        -p alvr_server_openvr \
        -p alvr_dashboard \
        -p alvr_vulkan_layer \
        -p alvr_vrcompositor_wrapper

    # Generate icons
    for res in 16x16 32x32 48x48 64x64 128x128 256x256; do
        mkdir -p "icons/hicolor/${res}/apps/"
        magick 'alvr/dashboard/resources/dashboard.ico' -thumbnail "${res}" -alpha on -background none -flatten "icons/hicolor/${res}/apps/alvr.png"
    done
}`

None of theese methods work. it is like it is completly blinded. Even if nvcc is installed
